(API) Salesforce Marketing Cloud - Changelog

mohan-h opened the pull request:
Initial commit

mohan-h committed:
Initial commit

vgiurgiu commented on help/destinations/catalog/email-marketing/salesforce-marketing-cloud-exact-target.md:
```diff
@@ -257,4 +257,17 @@ All [!DNL Adobe Experience Platform] destinations are compliant with data usage
+* [!DNL Salesforce Marketing Cloud] [documentation](https://developer.salesforce.com/docs/marketing/marketing-cloud/guide/updateContacts.html) explaining how contacts are updated with the specified information in the specified attribute groups.
+
+## Changelog
+
```
Let's introduce this section with an intro paragraph: 

"This section captures the functionality and significant documentation updates made to this destination connector"

vgiurgiu commented on help/destinations/catalog/email-marketing/salesforce-marketing-cloud-exact-target.md:
```diff
@@ -257,4 +257,17 @@ All [!DNL Adobe Experience Platform] destinations are compliant with data usage
+| Feb 08, 2023 | 1.0 | **Changed** <ul><li>Documentation: Back-link reference anchor to gather credentials, `PLATIR-25925`</li> <li>Documentation: Corrected Markdown formatting used in the custom domain explanation</li></ul> **Fixed** <ul><li>Connector: Removed extraneous comma from pebble template</li></ul> |
+| Jan 30, 2023 | 1.0 | **Changed** <ul><li>Documentation: Additional guidance on the mapping field value to be specified during the Scheduling step of the workflow</li> <li>Documentation: Data type and screenshots, `PLATIR-25602`</li></ul> |
+| Oct 11, 2022 | 1.0 | **Added** <ul><li>Go-Live</li></ul> |
+| Sep 22, 2022 | 1.0 | **Added** <ul><li>Public Beta</li></ul> |
```
Where did you get the public beta information from? I'm not aware that the destination was ever in this stage.

mohan-h replied:
We had pushed the connector in Sep-22, Subsequently there were functionality changes and it was additionally updated in Aug-22. Unfortunately, we have not been made privy to the dates when the connector / documentation is released externally, similarly when connector / documentation updates are released the information dissemination is not reaching back to us. 

See https://github.com/AdobeDocs/experience-platform.en/pull/88#issuecomment-1201501064 it went live and then again we updated through https://github.com/AdobeDocs/experience-platform.en/pull/113 hence had to assume the earlier release was probably a beta.

vgiurgiu replied:
I see. Please note that the destinations always go live at the same time that we publish the documentation. This coincides with the moment when the documentation PRs get closed and when you see this message from Adobe bot:

![Screen Shot 2023-04-10 at 12 23 44 PM](https://user-images.githubusercontent.com/19704189/230945356-735d9bd0-ba69-4106-ba6f-cb3a831f78f4.png)

vgiurgiu commented on help/destinations/catalog/email-marketing/salesforce-marketing-cloud-exact-target.md:
```diff
@@ -257,4 +257,17 @@ All [!DNL Adobe Experience Platform] destinations are compliant with data usage
+
+| Release Date | Connector Version | Description |
+| --- | --- | --- |
+| Feb 21, 2023 | 1.0 | **Changed** <ul><li>Documentation: Reference link to [!DNL Salesforce Marketing Cloud Account Engagement] as a mandatory subscription</li></ul> |
```
I don't think we've agreed as a team yet on how to do versioning for the connectors, so I would suggest leaving this column out for now.

vgiurgiu replied:
Documentation update:

We added a reference link to <insert which section> to call out that  [!DNL Salesforce Marketing Cloud Account Engagement] is a mandatory subscription to use this destination.

vgiurgiu commented on help/destinations/catalog/email-marketing/salesforce-marketing-cloud-exact-target.md:
```diff
@@ -257,4 +257,17 @@ All [!DNL Adobe Experience Platform] destinations are compliant with data usage
+| Feb 21, 2023 | 1.0 | **Changed** <ul><li>Documentation: Reference link to [!DNL Salesforce Marketing Cloud Account Engagement] as a mandatory subscription</li></ul> |
+| Feb 17, 2023 | 1.0 | **Changed** <ul><li>Connector: Added the *Expires In* header to the authentication payload as it is supported by [!DNL Salesforce Marketing Cloud].</li> <li>Connector: Updated Pebble template to include a hard-coded *Email Demographics* mapping at the end of the body, `PLATIR-27068`</li></ul> **Fixed** <ul><li>Connector: Corrected domain name regex.</li></ul> |
+| Feb 08, 2023 | 1.0 | **Changed** <ul><li>Documentation: Back-link reference anchor to gather credentials, `PLATIR-25925`</li> <li>Documentation: Corrected Markdown formatting used in the custom domain explanation</li></ul> **Fixed** <ul><li>Connector: Removed extraneous comma from pebble template</li></ul> |
+| Jan 30, 2023 | 1.0 | **Changed** <ul><li>Documentation: Additional guidance on the mapping field value to be specified during the Scheduling step of the workflow</li> <li>Documentation: Data type and screenshots, `PLATIR-25602`</li></ul> |
```
Rephrasing suggestion to make this more valuable to the customer: 

Documentation update (PLATIR-25602): 

* https://experienceleague.adobe.com/docs/experience-platform/destinations/catalog/email-marketing/salesforce-marketing-cloud-exact-target.html?lang=en#prerequisites-destination  -> We updated the prerequisites in Salesforce section to call out that attributes need to be created on the Salesforce side. This section now includes detailed instructions on how to do that and best practices around naming the attributes in Salesforce.
* https://experienceleague.adobe.com/docs/experience-platform/destinations/catalog/email-marketing/salesforce-marketing-cloud-exact-target.html?lang=en#schedule-segment-export-example  -> We added clear instructions on how to use the Mapping ID for each activated segment

vgiurgiu commented on help/destinations/catalog/email-marketing/salesforce-marketing-cloud-exact-target.md:
```diff
@@ -257,4 +257,17 @@ All [!DNL Adobe Experience Platform] destinations are compliant with data usage
+| Feb 17, 2023 | 1.0 | **Changed** <ul><li>Connector: Added the *Expires In* header to the authentication payload as it is supported by [!DNL Salesforce Marketing Cloud].</li> <li>Connector: Updated Pebble template to include a hard-coded *Email Demographics* mapping at the end of the body, `PLATIR-27068`</li></ul> **Fixed** <ul><li>Connector: Corrected domain name regex.</li></ul> |
+| Feb 08, 2023 | 1.0 | **Changed** <ul><li>Documentation: Back-link reference anchor to gather credentials, `PLATIR-25925`</li> <li>Documentation: Corrected Markdown formatting used in the custom domain explanation</li></ul> **Fixed** <ul><li>Connector: Removed extraneous comma from pebble template</li></ul> |
+| Jan 30, 2023 | 1.0 | **Changed** <ul><li>Documentation: Additional guidance on the mapping field value to be specified during the Scheduling step of the workflow</li> <li>Documentation: Data type and screenshots, `PLATIR-25602`</li></ul> |
+| Oct 11, 2022 | 1.0 | **Added** <ul><li>Go-Live</li></ul> |
```
Let's rephrase to "initial destination release and documentation publish."

vgiurgiu commented on help/destinations/catalog/email-marketing/salesforce-marketing-cloud-exact-target.md:
```diff
@@ -257,4 +257,17 @@ All [!DNL Adobe Experience Platform] destinations are compliant with data usage
+| --- | --- | --- |
+| Feb 21, 2023 | 1.0 | **Changed** <ul><li>Documentation: Reference link to [!DNL Salesforce Marketing Cloud Account Engagement] as a mandatory subscription</li></ul> |
+| Feb 17, 2023 | 1.0 | **Changed** <ul><li>Connector: Added the *Expires In* header to the authentication payload as it is supported by [!DNL Salesforce Marketing Cloud].</li> <li>Connector: Updated Pebble template to include a hard-coded *Email Demographics* mapping at the end of the body, `PLATIR-27068`</li></ul> **Fixed** <ul><li>Connector: Corrected domain name regex.</li></ul> |
+| Feb 08, 2023 | 1.0 | **Changed** <ul><li>Documentation: Back-link reference anchor to gather credentials, `PLATIR-25925`</li> <li>Documentation: Corrected Markdown formatting used in the custom domain explanation</li></ul> **Fixed** <ul><li>Connector: Removed extraneous comma from pebble template</li></ul> |
```
same comment as above about the Pebble template. The real question is why the comma was removed from the Pebble template. if you're not aware of what exactly this comma removal fixed, we can leave this row out altogether from the docs.

vgiurgiu reviewed (commented):
I left a few comments. The general high-level feedback that I wanted to share is about the perspective that we should use in the changelog. 

Since the changelog is customer-facing, I don't think it should be a line-by-line record of all the changes we made, but rather highlighting the customer value in the changes that are transparent to the customer. 

let me know please if you have any questions. happy to discuss.

mohan-h commented:
> I left a few comments. The general high-level feedback that I wanted to share is about the perspective that we should use in the changelog.
> 
> Since the changelog is customer-facing, I don't think it should be a line-by-line record of all the changes we made, but rather highlighting the customer value in the changes that are transparent to the customer.
> 
> let me know please if you have any questions. happy to discuss.

As per the suggestion to include suffix `update` between `Documentation` and `:`. 
After adding `update` the headers would read as below
```
Changed
<ul>
    <li> Documentation update: .... </li>
</ul>
```

Having both `Changed` and `update` reads superfluous.

So does your recommendation also imply I have to remove the `Changed`, `Added` and `Fixed` breakup and only have bullets which start with `Documentation update:` or `Connector update:`?

Also have changed the layout altogether, hope this is preferable.

Also added the collapse section around Changelog but it doesn't reflect on got preview, please could you guide.

mohan-h committed:
Updated changelog layout and content

mohan-h committed:
Collapsible sections

vgiurgiu reviewed (commented):
Thanks for the recent updates. We're getting closer to being able to publish. See below how I envision the changelog section (including collapsible sections):

```



### Changelog

This section captures the functionality and significant documentation updates made to this destination connector.

+++ View changelog

|Release month|Update type|Description|
|---|---|---|
|February 2023|Documentation update|We updated the [Prerequisites in (API) Salesforce Marketing Cloud](#prerequisites-destination) section to include a reference link calling out that [!DNL Salesforce Marketing Cloud Account Engagement] is a mandatory subscription to use this destination.|
|February 2023|Functionality update|We fixed an issue where an incorrect configuration in the destination was causing a malformed JSON to be sent to Salesforce. This resulted in some users seeing high numbers of identities failed on activation. (PLATIR-26299)|
|January 2023|Documentation update| <ul><li>We updated the [[Prerequisites in [!DNL Salesforce]](#prerequisites-destination) section to call out that attributes need to be created on the [!DNL Salesforce] side. This section now includes detailed instructions on how to do that and best practices around naming the attributes in [!DNL Salesforce]. (PLATIR-25602)</li><li>We added clear instructions on how to use the Mapping ID for each activated segment in the [segment scheduling](#schedule-segment-export-example) step. (PLATIR-25602)</li></ul> |
|October 2022|Initial release|Initial destination release and documentation publish.|

{style="table-layout:auto"}

+++

```

Let me know your thoughts but this is in my mind how we'd like to shape the changelog for this and future destinations moving forward. 

Note also that you won't be able to see the collapsible sections preview in the Git previewer since this is a custom Adobe markdown implementation, so Git is unaware of it. 


Do make sure please that you are using the Adobe Markdown Authoring extension for VS Code when you author docs: https://marketplace.visualstudio.com/items?itemName=AdobeExL.adobe-markdown-authoring&ssr=false#overview

I'll submit an enhancement request so that an Adobe team implements collapsible sections preview in the extension.

mohan-h commented:
Sure, the initial table layout works. Also, shall we tack on `PLATIR-27068` and PLATIR-26299 together? I didn't have `PLATIR-26299` as the dev team had shared `PLATIR-27068`. Have pushed the change per your preference.

mohan-h committed:
Reverted changelog to table

Reverted changelog to table, updated content per Adobe review recommendation

vgiurgiu commented on help/destinations/catalog/email-marketing/salesforce-marketing-cloud-exact-target.md:
```diff
@@ -257,4 +257,26 @@ All [!DNL Adobe Experience Platform] destinations are compliant with data usage
+
+{style="table-layout:auto"}
+
+<!--
```
I think that we can remove the commented-out line. I'll do that after I copy over the PR into internal git.
